hw/ip/lc_ctrl/data/lc_ctrl.hjson

```diff
@@ -999,6 +1028,32 @@
       ],
     }
 
+    ///////////////////////
+    // Hardware Revision //
+    ///////////////////////
+    { name:     "HW_REV",
+      desc:     '''
+                This register holds the 32bit hardware revision,
+                which is split into two 16bit fields holding the
+                !!CHIP_GEN and !!CHIP_REV numbers.
+                '''
+      swaccess: "ro",
+      hwaccess: "hwo",
+      hwext:    "true",
+      tags: [ // The value of this register is fixed and determined by HW.
+              "excl:CsrAllTests:CsrExclCheck"],
+      fields: [
+        { bits: "16+HwRevFieldWidth-1:16",
```

Comment:
ah okay. So we allocate 16 bits to each field, and then let the parameter pick the final width internally?

Reply:
Yeah this was just a way to use a shared parameter for the width instead of using literals everywhere. I can also just hardcode this to 16 everywhere...